libraries/node-core-library/src/LockFile.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as path from 'path';
+import * as child_process from 'child_process';
+
+/**
+ * Helper function that is exported for unit tests only
+ */
+export function getProcessStartTime(pid: string): string | undefined {
+  let args: string[];
+  if (process.platform === 'darwin') {
+    args = [`-p ${pid}`, '-o lstart'];
+  } else if (process.platform === 'linux') {
+    args = ['-p', pid.toString(), '-o', 'lstart'];
+  } else {
+    throw new Error(`Unsupported system: ${process.platform}`);
+  }
+
+  const psResult: string = child_process.spawnSync('ps', args).stdout.toString();
+  const psSplit: string[] = psResult.split('\n');
+  return psSplit[1] ? psSplit[1].trimRight().trimLeft() : undefined;
```

Comment:
>psSplit[1].trimRight().trimLeft() [](start = 22, length = 33)

you can do trim() instead of trimRight().trimLeft()

Reply:
Also as a sanity check, maybe make sure that the trimmed string is at least a certain number of characters long

---
In reply to: [165799122](https://github.com/Microsoft/web-build-tools/pull/517#discussion_r165799122) [](ancestors = 165799122)